src/pudl/output/pudltabl.py

```diff
@@ -1127,32 +1146,64 @@ def gens_mega_eia(self, update: bool = False) -> pd.DataFrame:
                     "The frequency of the pudl_out object must be `AS` for the "
                     f"plant-parts table and we got {self.freq}"
                 )
+            if gens_cols is None:
```

Comment:
nice! I like seeing this `gens_cols` in action, should we require these columns to be passed in no matter what?